src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -18,60 +17,85 @@
+            if (String.Equals(Environment.GetEnvironmentVariable(EnvironmentVariableNames.DISABLE_PUBLISH_AND_PACK_RELEASE), "true", StringComparison.OrdinalIgnoreCase))
+            {
+                return Enumerable.Empty<string>();
+            }
 
-            if (parseResult.HasOption(configOption) || globalProperties.ContainsKey(MSBuildPropertyNames.CONFIGURATION))
-                yield break;
+            var globalProperties = GetGlobalPropertiesFromUserArgs();
 
-            // CLI Configuration values take precedence over ones in the project. Passing PublishRelease as a global property allows it to take precedence.
-            project = GetTargetedProject(slnOrProjectArgs, globalProperties, defaultedConfigurationProperty);
+            // Configuration doesn't work in a .proj file, but it does as a global property.
+            // Detect either A) --configuration option usage OR /p:Configuration=Foo, if so, don't use these properties.
+            if (_parseResult.HasOption(_configOption) || globalProperties.ContainsKey(MSBuildPropertyNames.CONFIGURATION))
+                return Enumerable.Empty<string>();
+
+            ProjectInstance project = GetTargetedProject(globalProperties);
 
             if (project != null)
             {
-                string configurationToUse = "";
-                string releasePropertyFlag = project.GetPropertyValue(defaultedConfigurationProperty);
-                if (!string.IsNullOrEmpty(releasePropertyFlag))
-                    configurationToUse = releasePropertyFlag.Equals("true", StringComparison.OrdinalIgnoreCase) ? MSBuildPropertyNames.CONFIGURATION_RELEASE_VALUE : "";
-
-                if (!string.IsNullOrEmpty(configurationToUse) && !ProjectHasUserCustomizedConfiguration(project, defaultedConfigurationProperty))
-                    yield return $"-property:{MSBuildPropertyNames.CONFIGURATION}={configurationToUse}";
+                string releasePropertyFlag = project.GetPropertyValue(_defaultedConfigurationProperty);
+                if (!string.IsNullOrEmpty(releasePropertyFlag)) // The project set PublishRelease or PackRelease itself.
+                {
+                    string configurationToUse = releasePropertyFlag.Equals("true", StringComparison.OrdinalIgnoreCase) ? MSBuildPropertyNames.CONFIGURATION_RELEASE_VALUE : MSBuildPropertyNames.CONFIGURATION_DEBUG_VALUE;
+                    return new List<string> {
+                        $"-property:{MSBuildPropertyNames.CONFIGURATION}={configurationToUse}",
+                        _isPublishingSolution ? $"-property:_SolutionLevel{_defaultedConfigurationProperty}=true" : "" // Allows us to spot conflicting configuration values during evaluation.
+                    };
+                }
+                else if (GetProjectMaxModernTargetFramework(project) >= 8) // For 8.0, these properties are enabled by default.
```

Comment:
That's a good point, sadly we couldn't think of a better alternative. @/baronfel and @/richlander and I discussed these 3 evils:

For PublishRelease and .NET 8, which of the following is a better choice?

1) When multitargeting, if net8.0 is in your set of tfms, the default for publish will be Release no matter the TFM
2) When multitargeting, Publish will always default to Debug
3) When using the 8.0 SDK, Release will be the default of Publish regardless of the TFM

We Chet said we didn't want 2. And given only about 5% of customers multi-target, I think 1) is better. 

Even if we handle the default value in targets I think it will be too late to pivot. We can add the default value but I don't think we can gate the behavior on TFM.

Reply:
Just for record keeping, now this has been changed and it should respect the `-f` option above all else. Multitargeting isn't even an issue with publish and pack will have it own by default.